From 0455d918688d9d208644d7f5f6f10d6199486e9c Mon Sep 17 00:00:00 2001
From: Al Stone <ahs3@redhat.com>
Date: Mon, 28 Sep 2020 12:49:23 -0600
Subject: [PATCH 32/40] Support SLIC and MSDM in a big-endian world

Signed-off-by: Al Stone <ahs3@redhat.com>
---
 source/common/dmtbdump3.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

Index: acpica-unix2-20200925/source/common/dmtbdump3.c
===================================================================
--- acpica-unix2-20200925.orig/source/common/dmtbdump3.c
+++ acpica-unix2-20200925/source/common/dmtbdump3.c
@@ -68,9 +68,10 @@ void
 AcpiDmDumpSlic (
     ACPI_TABLE_HEADER       *Table)
 {
+    UINT32                  TableLength = AcpiUtReadUint32(&Table->Length);
 
-    (void) AcpiDmDumpTable (Table->Length, sizeof (ACPI_TABLE_HEADER), Table,
-        Table->Length - sizeof (*Table), AcpiDmTableInfoSlic);
+    (void) AcpiDmDumpTable (TableLength, sizeof (ACPI_TABLE_HEADER), Table,
+        TableLength - sizeof (*Table), AcpiDmTableInfoSlic);
 }
 
 
