diff -Naur acpica-unix2-20150717/tests/aapits/Makefile acpica-unix2-20150717-aapits/tests/aapits/Makefile
--- acpica-unix2-20150717/tests/aapits/Makefile	2015-07-17 10:05:11.000000000 -0600
+++ acpica-unix2-20150717-aapits/tests/aapits/Makefile	2015-08-05 10:20:14.776322314 -0600
@@ -31,6 +31,7 @@
 	../../source/components/debugger/dbinput.c \
 	../../source/components/debugger/dbmethod.c \
 	../../source/components/debugger/dbnames.c \
+	../../source/components/debugger/dbobject.c \
 	../../source/components/debugger/dbstats.c \
 	../../source/components/debugger/dbtest.c \
 	../../source/components/debugger/dbutils.c \
@@ -39,7 +40,6 @@
 	../../source/components/disassembler/dmcstyle.c \
 	../../source/components/disassembler/dmdeferred.c \
 	../../source/components/disassembler/dmnames.c \
-	../../source/components/disassembler/dmobject.c \
 	../../source/components/disassembler/dmopcode.c \
 	../../source/components/disassembler/dmresrc.c \
 	../../source/components/disassembler/dmresrcl.c \
@@ -71,6 +71,7 @@
 	../../source/components/hardware/hwxfsleep.c \
 	../../source/components/dispatcher/dsargs.c \
 	../../source/components/dispatcher/dscontrol.c \
+	../../source/components/dispatcher/dsdebug.c \
 	../../source/components/dispatcher/dsfield.c \
 	../../source/components/dispatcher/dsinit.c \
 	../../source/components/dispatcher/dsmethod.c \
@@ -183,6 +184,7 @@
 	../../source/components/utilities/utmath.c \
 	../../source/components/utilities/utmisc.c \
 	../../source/components/utilities/utmutex.c \
+	../../source/components/utilities/utnonansi.c \
 	../../source/components/utilities/utobject.c \
 	../../source/components/utilities/utosi.c \
 	../../source/components/utilities/utownerid.c \
