Index: acpica-unix-20150408/tests/aapits/Makefile
===================================================================
--- acpica-unix-20150408.orig/tests/aapits/Makefile
+++ acpica-unix-20150408/tests/aapits/Makefile
@@ -47,6 +47,7 @@ SRCS=	atexec.c atmain.c \
 	../../source/components/disassembler/dmresrcs.c \
 	../../source/components/disassembler/dmutils.c \
 	../../source/components/disassembler/dmwalk.c \
+	../../source/components/disassembler/dmcstyle.c \
 	../../source/components/events/evevent.c \
 	../../source/components/events/evgpe.c \
 	../../source/components/events/evgpeblk.c \
