Index: acpica-unix-20140926/tests/aapits/Makefile
===================================================================
--- acpica-unix-20140926.orig/tests/aapits/Makefile
+++ acpica-unix-20140926/tests/aapits/Makefile
@@ -16,6 +16,7 @@ SRCS=	atexec.c atmain.c \
 	atosxfwrap.c \
 	osunixxf.c \
 	../../source/common/ahids.c \
+	../../source/common/ahuuids.c \
 	../../source/common/cmfsize.c \
 	../../source/common/getopt.c \
 	../../source/components/hardware/hwtimer.c \
@@ -174,6 +175,7 @@ SRCS=	atexec.c atmain.c \
 	../../source/components/utilities/utexcep.c \
 	../../source/components/utilities/utfileio.c \
 	../../source/components/utilities/utglobal.c \
+	../../source/components/utilities/uthex.c \
 	../../source/components/utilities/utids.c \
 	../../source/components/utilities/utinit.c \
 	../../source/components/utilities/utlock.c \
@@ -189,6 +191,7 @@ SRCS=	atexec.c atmain.c \
 	../../source/components/utilities/utstate.c \
 	../../source/components/utilities/utstring.c \
 	../../source/components/utilities/uttrack.c \
+	../../source/components/utilities/utuuid.c \
 	../../source/components/utilities/utxface.c \
 	../../source/components/utilities/utxferror.c \
 	../../source/components/utilities/utxfinit.c \
