Index: acpica-unix-20180508/Makefile
===================================================================
--- acpica-unix-20180508.orig/Makefile
+++ acpica-unix-20180508/Makefile
@@ -20,3 +20,22 @@ else
 	include generate/unix/Makefile.config
 	include generate/unix/Makefile.common
 endif
+
+check:
+	# ASL tests
+	$(CURDIR)/tests/aslts.sh $(CURDIR)/tests/aslts $(CURDIR)/generate/unix
+
+	# misc tests
+	$(CURDIR)/debian/run-misc-tests.sh $(CURDIR) 20180810
+
+	# Template tests
+	cd $(CURDIR)/tests/templates
+	make
+	if [ -f diff.log ] ; \
+	then \
+		if [ -s diff.log ] ; \
+		then \
+			exit 1		# implies errors occurred ; \
+		fi ; \
+	fi
+
