Index: acpica-unix-20160422/source/compiler/asllookup.c
===================================================================
--- acpica-unix-20160422.orig/source/compiler/asllookup.c
+++ acpica-unix-20160422/source/compiler/asllookup.c
@@ -119,6 +119,7 @@ LkIsObjectUsed (
 {
     ACPI_NAMESPACE_NODE     *Node = ACPI_CAST_PTR (ACPI_NAMESPACE_NODE, ObjHandle);
     ACPI_NAMESPACE_NODE     *Next;
+    ACPI_NAME_UNION         tmp, tmp2;
     ASL_METHOD_LOCAL        *MethodLocals;
     ASL_METHOD_LOCAL        *MethodArgs;
     UINT32                  i;
@@ -228,8 +229,10 @@ LkIsObjectUsed (
              * Issue a remark even if it is a reserved name (starts
              * with an underscore).
              */
+            ACPI_MOVE_32_TO_32(&tmp.Ascii, Node->Name.Ascii);
+            ACPI_MOVE_32_TO_32(&tmp2.Ascii, Next->Name.Ascii);
             sprintf (MsgBuffer, "Name [%4.4s] is within a method [%4.4s]",
-                Node->Name.Ascii, Next->Name.Ascii);
+                tmp.Ascii, tmp2.Ascii);
             AslError (ASL_REMARK, ASL_MSG_NOT_REFERENCED,
                 LkGetNameOp (Node->Op), MsgBuffer);
             return (AE_OK);
