Index: acpica-unix-20180508/source/compiler/dttable2.c
===================================================================
--- acpica-unix-20180508.orig/source/compiler/dttable2.c
+++ acpica-unix-20180508/source/compiler/dttable2.c
@@ -1797,6 +1797,7 @@ DtCompileTpm2 (
     ACPI_TABLE_TPM2         *Tpm2Header;
     DT_SUBTABLE             *ParentTable;
     ACPI_STATUS             Status = AE_OK;
+    UINT32		    Tmp32;
 
 
     /* Compile the main table */
@@ -1831,7 +1832,8 @@ DtCompileTpm2 (
 
     /* Subtable type depends on the StartMethod */
 
-    switch (Tpm2Header->StartMethod)
+    ACPI_MOVE_32_TO_32(&Tmp32, &Tpm2Header->StartMethod);
+    switch (Tmp32)
     {
     case ACPI_TPM2_COMMAND_BUFFER_WITH_ARM_SMC:
 
@@ -1861,16 +1863,14 @@ DtCompileTpm2 (
     case ACPI_TPM2_RESERVED9:
     case ACPI_TPM2_RESERVED10:
 
-        AcpiOsPrintf ("\n**** Reserved TPM2 Start Method type 0x%X\n",
-            Tpm2Header->StartMethod);
+        AcpiOsPrintf ("\n**** Reserved TPM2 Start Method type 0x%X\n", Tmp32);
         Status = AE_ERROR;
         break;
 
     case ACPI_TPM2_NOT_ALLOWED:
     default:
 
-        AcpiOsPrintf ("\n**** Unknown TPM2 Start Method type 0x%X\n",
-            Tpm2Header->StartMethod);
+        AcpiOsPrintf ("\n**** Unknown TPM2 Start Method type 0x%X\n", Tmp32);
         Status = AE_ERROR;
         break;
     }
