diff -Naur acpica-unix2-20191213.orig/source/tools/acpiexec/aemain.c acpica-unix2-20191213/source/tools/acpiexec/aemain.c
--- acpica-unix2-20191213.orig/source/tools/acpiexec/aemain.c	2020-02-23 16:32:25.935853162 -0700
+++ acpica-unix2-20191213/source/tools/acpiexec/aemain.c	2020-02-23 16:44:38.949901103 -0700
@@ -84,7 +84,6 @@
 UINT8                       AcpiGbl_RegionFillValue = 0;
 BOOLEAN                     AcpiGbl_IgnoreErrors = FALSE;
 BOOLEAN                     AcpiGbl_AbortLoopOnTimeout = FALSE;
-BOOLEAN                     AcpiGbl_DbOpt_NoRegionSupport = FALSE;
 UINT8                       AcpiGbl_UseHwReducedFadt = FALSE;
 BOOLEAN                     AcpiGbl_DoInterfaceTests = FALSE;
 BOOLEAN                     AcpiGbl_LoadTestTables = FALSE;
@@ -549,6 +548,10 @@
         goto NormalExit;
     }
 
+    /* Initialize globals modified by command line options */
+
+    AcpiGbl_DbOpt_NoRegionSupport = FALSE;
+
     /* Get the command line options */
 
     ExitCode = AeDoOptions (argc, argv);
