Index: acpica-unix-20200925/source/include/platform/aclinux.h
===================================================================
--- acpica-unix-20200925.orig/source/include/platform/aclinux.h
+++ acpica-unix-20200925/source/include/platform/aclinux.h
@@ -217,7 +217,7 @@
 
 #if defined(__ia64__)    || (defined(__x86_64__) && !defined(__ILP32__)) ||\
     defined(__aarch64__) || defined(__PPC64__) ||\
-    defined(__s390x__) ||\
+    defined(__s390x__) || defined(__mips64) ||\
     (defined(__riscv) && (defined(__LP64__) || defined(_LP64)))
 #define ACPI_MACHINE_WIDTH          64
 #define COMPILER_DEPENDENT_INT64    long
