Index: acpica-unix-20200214/source/compiler/dtcompiler.h
===================================================================
--- acpica-unix-20200214.orig/source/compiler/dtcompiler.h
+++ acpica-unix-20200214/source/compiler/dtcompiler.h
@@ -353,7 +353,7 @@ DtCreateTableUnit (
 
 /* dtparser - lex/yacc files */
 
-UINT64                      DtCompilerParserResult; /* Expression return value */
+extern UINT64                      DtCompilerParserResult; /* Expression return value */
 int
 DtCompilerParserparse (
     void);
Index: acpica-unix-20200214/source/compiler/aslcompiler.l
===================================================================
--- acpica-unix-20200214.orig/source/compiler/aslcompiler.l
+++ acpica-unix-20200214/source/compiler/aslcompiler.l
@@ -48,7 +48,7 @@
 
 #include <stdlib.h>
 #include <string.h>
-YYSTYPE AslCompilerlval;
+extern YYSTYPE AslCompilerlval;
 
 /*
  * Generation:  Use the following command line:
Index: acpica-unix-20200214/source/compiler/prparser.l
===================================================================
--- acpica-unix-20200214.orig/source/compiler/prparser.l
+++ acpica-unix-20200214/source/compiler/prparser.l
@@ -116,7 +116,7 @@ Identifier      [a-zA-Z][0-9a-zA-Z]*
 /*
  * Local support functions
  */
-YY_BUFFER_STATE         LexBuffer;
+extern YY_BUFFER_STATE         LexBuffer;
 
 
 /******************************************************************************
