diff -Naur acpica-unix2-20200214.orig/source/compiler/aslcompiler.l acpica-unix2-20200214/source/compiler/aslcompiler.l
--- acpica-unix2-20200214.orig/source/compiler/aslcompiler.l	2020-02-25 17:24:22.169290551 -0700
+++ acpica-unix2-20200214/source/compiler/aslcompiler.l	2020-02-25 17:26:01.807138585 -0700
@@ -584,7 +584,7 @@
 "IPMI"                      { count (0); return (PARSEOP_REGIONSPACE_IPMI); }
 "GeneralPurposeIo"          { count (0); return (PARSEOP_REGIONSPACE_GPIO); }       /* ACPI 5.0 */
 "GenericSerialBus"          { count (0); return (PARSEOP_REGIONSPACE_GSBUS); }      /* ACPI 5.0 */
-"PlatformCommChannel"       { count (0); return (PARSEOP_REGIONSPACE_PCC); }        /* ACPI 5.0 */
+"PCC"                       { count (0); return (PARSEOP_REGIONSPACE_PCC); }        /* ACPI 5.0 */
 "FFixedHW"                  { count (0); return (PARSEOP_REGIONSPACE_FFIXEDHW); }
 
     /* ResourceTypeKeyword: Resource Usage - Resource Descriptors */
diff -Naur acpica-unix2-20200214.orig/source/components/utilities/utdecode.c acpica-unix2-20200214/source/components/utilities/utdecode.c
--- acpica-unix2-20200214.orig/source/components/utilities/utdecode.c	2020-02-14 10:33:55.000000000 -0700
+++ acpica-unix2-20200214/source/components/utilities/utdecode.c	2020-02-25 17:26:28.524365981 -0700
@@ -116,7 +116,7 @@
     "IPMI",              /* 0x07 */
     "GeneralPurposeIo",  /* 0x08 */
     "GenericSerialBus",  /* 0x09 */
-    "PlatformCommChannel"/* 0x0A */
+    "PCC"                /* 0x0A */
 };
 
 
